src/Features/Core/Portable/StackTraceExplorer/StackTraceExplorerService.cs

```diff
@@ -87,11 +84,16 @@ private static ImmutableArray<Document> GetFileMatches(Solution solution, StackF
         lineNumber = int.Parse(lineString);
 
         var documentName = Path.GetFileName(fileName);
-        var potentialMatches = new HashSet<Document>();
+        var potentialMatches = new HashSet<TextDocument>();
 
         foreach (var project in solution.Projects)
         {
-            foreach (var document in project.Documents)
+            // As of writing there is no way to get all the documents for a specific project
+            // so we need to check both the main and additional documents. If more document types
+            // get added this likely will need to be updated.
+            var allDocuments = project.Documents.Concat(project.AdditionalDocuments);
+
+            foreach (var document in allDocuments)
             {
                 if (document.FilePath == fileName)
```

Comment:
If matching by an exact file path is always the winner, shouldn't we be calling [GetDocumentIdsWithFilePath](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.solution.getdocumentidswithfilepath?view=roslyn-dotnet-4.9.0) and only if it doesn't find anything, then falling back to the weaker search?

Reply:
Considering it is returning the first document, I suggested `GetFirstDocumentIdWithFilePath`

Reply:
`GetFirstDocumentIdWithFilePath` this appears to be only on `ProjectState`. I don't see an equivalent for `Solution`